doc/nondeterministic.md

```diff
@@ -45,16 +44,13 @@ class MyTests : ShouldSpec() {
 
 #### Exceptions
 
-By default, `eventually` will ignore any exception that is thrown inside the function (note, that means it won't catch `Error`).
-If you want to be more specific, you can tell `eventually` to ignore specific exceptions and any others will immediately fail the test.
+By default, `eventually` will ignore any AssertionError thrown inside the function,
+while any others will then immediately fail the test.
 
-Let's assume that our example from before throws a `UserNotFoundException` while the user is not found in the database.
+You can also tell `eventually` to ignore other exceptions:
+Let's assume that our example from before throws a `UserNotFoundException` when the user is not found in the database.
 It will eventually return the user when the message is processed by the system.
-
-In this scenario, we can explicitly skip the exception that we expect to happen until the test passed, but any other exceptions would
```

Comment:
I'm not sure why you deleted this block of text

Reply:
it was quite convoluted, I tried to shorten it to improve comprehensibility. Also, I had to adjust it to the slight functional changes due to the previous inconsistencies.